cmake/modules/kconfig.cmake

```diff
@@ -147,10 +147,10 @@ set(EXTRA_KCONFIG_TARGET_COMMAND_FOR_hardenconfig
   ${ZEPHYR_BASE}/scripts/kconfig/hardenconfig.py
   )
 
+set_ifndef(KCONFIG_TARGETS menuconfig guiconfig hardenconfig)
+
 foreach(kconfig_target
-    menuconfig
-    guiconfig
-    hardenconfig
+    ${KCONFIG_TARGETS}
```

Comment:
This patch also leaves me vaguely nervous about maintainability for the same reasons as the previous one.

Can the necessary features required by sysbuild be extracted into helper functions somehow and shared with the zephyr build system, instead of just executing this module like this?

Otherwise it may become difficult to modify the zephyr-specific portions without accidentally breaking sysbuild.

Reply:
> Can the necessary features required by sysbuild be extracted into helper functions somehow and shared with the zephyr build system, instead of just executing this module like this?

I understand your concern, but I consider this is part of the continues effort of #41301, as described here:
https://github.com/zephyrproject-rtos/zephyr/pull/40555#discussion_r825786952

There is of course a fine balance on what belongs inside the SYSBuild PR and what is a continuation of #41301.
I judged this to be part of #41301 continuation.